core/server/proxy/src/main/java/alluxio/proxy/s3/signature/AWSSignatureProcessor.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.proxy.s3.signature;
+
+import alluxio.proxy.s3.S3ErrorCode;
+import alluxio.proxy.s3.S3Exception;
+import alluxio.proxy.s3.auth.AWSAuthInfo;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.MultivaluedMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Parser to process AWS V2 & V4 auth request. Creates string to sign and auth
+ * header. For more details refer to AWS documentation https://docs.aws
+ * .amazon.com/general/latest/gr/sigv4-create-canonical-request.html.
+ **/
+
+public class AWSSignatureProcessor implements SignatureProcessor {
+
+  private static final Logger LOG =
+            LoggerFactory.getLogger(AWSSignatureProcessor.class);
+
+  private ContainerRequestContext mContext;
+
+  /**
+   * Create a new {@link AWSSignatureProcessor}.
+   *
+   * @param context ContainerRequestContext
+   */
+  public AWSSignatureProcessor(ContainerRequestContext context) {
+    mContext = context;
+  }
+
+  @Override
+  public SignatureInfo parseSignature() throws S3Exception {
+    LowerCaseKeyStringMap headers =
+        LowerCaseKeyStringMap.fromHeaderMap(mContext.getHeaders());
+
+    String authHeader = headers.get(SignerConstants.AUTHORIZATION);
+
+    List<SignatureParser> signatureParsers = new ArrayList<>();
+    signatureParsers.add(new AuthorizationV4HeaderParser(authHeader,
+                headers.get(SignerConstants.X_AMZ_DATE)));
+    signatureParsers.add(new AuthorizationV4QueryParser(
+        StringToSignProducer.fromMultiValueToSingleValueMap(
+                mContext.getUriInfo().getQueryParameters())));
+    signatureParsers.add(new AuthorizationV2HeaderParser(authHeader));
+
+    SignatureInfo signatureInfo = null;
+    for (SignatureParser parser : signatureParsers) {
+      signatureInfo = parser.parseSignature();
+      if (signatureInfo != null) {
+        break;
+      }
+    }
```

Comment:
Because when the parse Signature function is called, it does not know whether the request information is passed through the header or param, but through all parsers trying to parse. So parsers are not used as  util class.